internal/header/feature_tag.go

```diff
@@ -0,0 +1,134 @@
+/* Copyright 2019 Freerware
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package header
+
+import (
+	"strconv"
+	"strings"
+)
+
+// FeatureTag represents a feature tag.
+// https://tools.ietf.org/html/rfc2295#section-6.1
+type FeatureTag string
+
+// Equals performs a case-insensitive comparison with the provided feature tag.
+// The US-ASCII charset is used for feature tag.
+func (t FeatureTag) Equals(tag FeatureTag) bool {
+	u, err := t.unquotedASCII()
+	if err != nil {
+		return false
+	}
+	q, err := t.quotedASCII()
+	if err != nil {
+		return false
+	}
+	uu, err := tag.unquotedASCII()
+	if err != nil {
+		return false
+	}
+	qq, err := tag.quotedASCII()
+	if err != nil {
+		return false
+	}
+	return strings.ToLower(u) == strings.ToLower(uu) ||
+		strings.ToLower(q) == strings.ToLower(qq)
+}
+
+func (t FeatureTag) quotedASCII() (string, error) {
```

Comment:
add comment for `quotedASCII(...)` as well as other types and methods in this file.

Reply:
done!